bee-message/src/unlock_block/mod.rs

```diff
@@ -119,7 +119,7 @@ fn verify_unlock_blocks<const VERIFY: bool>(unlock_blocks: &[UnlockBlock]) -> Re
             UnlockBlock::Reference(reference) => {
                 if index == 0
                     || reference.index() >= index as u16
```

Comment:
Maybe we don't need the cast here because index is already declared as `(u16..)`.

```suggestion
                    || reference.index() >= index
```

Reply:
Yeah we don't